src/inmanta/deploy/persistence.py

```diff
@@ -305,7 +306,8 @@ def error_and_log(message: str, **context: Any) -> None:
                     last_deployed_version=resource_id_parsed.version,
                     last_deployed_attribute_hash=resource.attribute_hash,
                     last_non_deploying_status=const.NonDeployingResourceState(status),
-                    **extra_fields,
+                    resource_status=state.ResourceStatus.UP_TO_DATE if status is const.ResourceState.deployed else None,
```

Comment:
I think, when we arrive here, the scheduler has already been updated, so we can just flush back the state we find there?

Reply:
Conceptually yes, implementation-wise it's a bit of a puzzle because we're outside the scheduler domain here. But I still that that that's what we should do, I'm just not sure yet how best to achieve it. Perhaps we consider this method the compatibility layer, and add a new method to this class for the new fields, one that can be fully driven by the scheduler rather than the deploy task.